pkg/networkmanager/manifest.json

```diff
@@ -39,5 +39,13 @@
                 }
             ]
         }
+    },
+    "config": {
+        "wireguard_package": {
+            "fedora": "wireguard-tools",
+            "arch": "wireguard-tools",
+            "ubuntu": "wireguard",
+            "debian": "wireguard"
```

Comment:
Debian also has the [wireguard-tools package](https://packages.debian.org/unstable/wireguard-tools). There is a [wireguard meta-package](https://packages.debian.org/unstable/wireguard) which depends on -tools and additionally the kernel modules; however, the standard Debian 12 (stable) kernel already includes them, so we dont need to be concerned about this any more. Installing "wireguard" is not wrong, but if it simplifies things, we can also hardcode "wireguard-tools". I also checked OpenSUSE, and it's [wireguard-tools](https://software.opensuse.org/package/wireguard-tools) there as well.

The main issue here is that RHEL/CentOS 8 does *not* have wireguard-tools. RHEL 9 does. So we can't add "rhel" here.

My proposal:
 * Drop this `config` bit in the manifest
 * In the code, check `os_release.PLATFORM_ID` and don't offer on-demand install for "platform:el8".
 * Otherwise, try to install `wireguard-tools`. So far the package name is the same on all our supported distributions.

FTR, I checked os-release on various images:

### rhel-8-9
```
NAME="Red Hat Enterprise Linux"    
VERSION="8.9 (Ootpa)"    
ID="rhel"    
ID_LIKE="fedora"    
VERSION_ID="8.9"    
PLATFORM_ID="platform:el8"    
``` 
    
### rhel-9-3
```
NAME="Red Hat Enterprise Linux"    
VERSION="9.3 (Plow)"    
ID="rhel"    
ID_LIKE="fedora"    
VERSION_ID="9.3"    
PLATFORM_ID="platform:el9"    
``` 
    
### ubuntu-stable
```
PRETTY_NAME="Ubuntu 23.04"    
NAME="Ubuntu"    
VERSION_ID="23.04"    
VERSION="23.04 (Lunar Lobster)"    
VERSION_CODENAME=lunar    
ID=ubuntu    
ID_LIKE=debian    
```

So at some point we can also check `.ID` and fall back to `.ID_LIKE`, to subsume RHEL, Ubuntu etc. into their parent distros. But that still leaves the RHEL 8 special case.